File: Project/C++/src/transaction.cpp
#include <unordered_set>
#include <string>
using namespace std;

class transaction
{
private:
    int tuple[10];

public:
    transaction(vector<string> line)
    {
        for (int i = 0; i < 10; i++)
        {
            tuple[i] = stoi(line[i]);
        }
    }

    int *getTuple()
    {
        return tuple;
    }

    string toString()
    {
        string res = "";
        for (int i = 0; i < 10; i++)
        {
            res.append(to_string(tuple[i]) + ";");
        }
        return res;
    }
};

File: Project/Java/Transaction.java
import java.util.Arrays;
import java.util.HashSet;

public class Transaction {

  private HashSet<Integer> classification;
  private int[] tuple;

  public Transaction(String[] line) {
    this.classification = new HashSet<>();

    this.tuple = new int[10];
    for (int i = 0; i < 10; i++) {
      this.tuple[i] = Integer.valueOf(line[i]);
    }
  }

  public void addClassification(int classification) {
    this.classification.add(classification);
  }

  public int[] getTuple() {
    return tuple;
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    for (int i = 0; i < tuple.length; i++) {
      sb.append(tuple[i] + ";");
    }
    return sb.toString();
  }
}
